pkg/master/master.go

```diff
@@ -132,7 +132,9 @@ type Config struct {
 	// ExternalHost is the host name to use for external (public internet) facing URLs (e.g. Swagger)
 	ExternalHost string
 
-	// If nil, the first result from net.InterfaceAddrs will be used.
+	// PublicAddress is the IP address where members of the cluster (kubelet,
+	// kube-proxy, services, etc.) can reach the master.
+	// If nil or 0.0.0.0, the host's default interface will be used.
```

Comment:
I kinda dislike that ChooseHostInterface() is getting called in multiple places now: the one you added in cmd/kube-apiserver/app/server.go and the existing one in pkg/master/master.go.  Is it safe to get rid of the one in master,go and just say that PublicAddress is required?  I wonder if that will break tests.